pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/DownloadParticipantListRoute.java

```diff
@@ -4,91 +4,115 @@
 import static org.broadinstitute.dsm.util.ElasticSearchUtil.MAX_RESULT_SIZE;
 
 import java.util.ArrayList;
-import java.util.Comparator;
+import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
-import java.util.TreeMap;
 
-import org.broadinstitute.dsm.export.ParticipantExcelGenerator;
-import org.broadinstitute.dsm.model.ParticipantColumn;
-import org.broadinstitute.dsm.model.elastic.export.excel.DownloadParticipantListPayload;
-import org.broadinstitute.dsm.model.elastic.export.excel.ParticipantRecordData;
-import org.broadinstitute.dsm.model.elastic.sort.Alias;
+import lombok.Getter;
+import lombok.Setter;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.model.Filter;
+import org.broadinstitute.dsm.model.elastic.export.tabular.ModuleExportConfig;
+import org.broadinstitute.dsm.model.elastic.export.tabular.TabularParticipantExporter;
+import org.broadinstitute.dsm.model.elastic.export.tabular.TabularParticipantParser;
 import org.broadinstitute.dsm.model.filter.FilterFactory;
 import org.broadinstitute.dsm.model.filter.Filterable;
+import org.broadinstitute.dsm.model.participant.ParticipantWrapperDto;
 import org.broadinstitute.dsm.model.participant.ParticipantWrapperResult;
 import org.broadinstitute.dsm.security.RequestHandler;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.statics.RoutePath;
 import org.broadinstitute.dsm.util.proxy.jackson.ObjectMapperSingleton;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import spark.QueryParamsMap;
 import spark.Request;
 import spark.Response;
 
 public class DownloadParticipantListRoute extends RequestHandler {
 
     private static final Logger logger = LoggerFactory.getLogger(DownloadParticipantListRoute.class);
 
-    public DownloadParticipantListRoute() {
+    /**
+     * Generates a file for download.  When removing the feature-flag-export-new role, processRequestNew should
+     * be renamed to processRequest, and the old 'processRequest' method should be deleted
+     */
+    public Object processRequest(Request request, Response response, String userId) throws Exception {
+        DownloadParticipantListPayloadNew payload =
+                ObjectMapperSingleton.instance().readValue(request.body(), DownloadParticipantListPayloadNew.class);
+        DownloadParticipantListParams params = new DownloadParticipantListParams(request.queryMap());
+
+        String realm = RoutePath.getRealm(request);
+        DDPInstance instance = DDPInstance.getDDPInstanceWithRole(realm, DBConstants.MEDICAL_RECORD_ACTIVATED);
```

Comment:
Before we do any download here or any more calls to db, I think we should check if the user is actually allowed to see the pt list and therefore allowed to download the data.
Have a look at `FilterRoute` how it checks the permissions there.
and I think we should also add an additional permission here `pt_list_download`. seeing that for FON we need that permission separated from "pt_list_view"